timeboost-types/src/decryption.rs

```diff
@@ -251,3 +427,103 @@ impl Subset {
         &self.bundles
     }
 }
+
+/// Accumulates resharing bundles (see `DkgAccumulator`).
+#[derive(Debug, Clone)]
+pub struct ResharingAccumulator {
+    store: KeyStore,
+    threshold: usize,
+    combkey: CombKey,
+    bundles: Vec<DkgBundle>,
+}
+
+impl ResharingAccumulator {
+    pub fn new(store: KeyStore, combkey: CombKey) -> Self {
+        Self {
+            threshold: store.committee().one_honest_threshold().get(),
+            store,
+            combkey,
+            bundles: Vec::new(),
+        }
+    }
+
+    pub fn committee(&self) -> &Committee {
+        &self.store.committee
+    }
+
+    pub fn bundles(&self) -> &[DkgBundle] {
+        &self.bundles
+    }
+
+    pub fn combkey(&self) -> &ThresholdCombKey {
+        &self.combkey
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.bundles.is_empty()
+    }
+
+    pub fn try_add(&mut self, bundle: DkgBundle) -> Result<(), VessError> {
+        // caller should ensure that no bundles are added after finalization
+        let Some(pub_share) = self.combkey.get_pub_share(bundle.origin().0.into()) else {
+            return Err(VessError::FailedVerification);
+        };
+
+        let aad: &[u8; 3] = b"dkg";
+        let committee = self.store.committee();
+        let vess = Vess::new_fast();
+        vess.verify_reshares(
```

Comment:
Unless performance has improved a lot I suggest to wrap this in a `tokio::task::block_in_place` so that the executor has a chance to move other tasks to different threads.